src/pickupManager.cpp

```diff
@@ -25,6 +25,14 @@ Pickup* PickupManager::randomPickup() {
 	effectNS::EffectData data;
 
 	data = effectTesting ? getPickupData(effectToTest) : effectDataNS::effectArray[getRandEffectArrIndex()];
+	if (data.effectType == effectNS::MYSTERY) {
+		effectNS::EffectData random = effectDataNS::effectArray[getRandEffectArrIndex()];
```

Comment:
Just use `randInt(0, effectDataNS::EFFECT_ARR_SIZE - 2` since mystery is the last effect. So no need for while loop